source/common/config/utility.cc

```diff
@@ -124,6 +125,38 @@ std::string Utility::resourceName(const ProtobufWkt::Any& resource) {
       fmt::format("Unknown type URL {} in DiscoveryResponse", resource.type_url()));
 }
 
+std::vector<Stats::TagExtractorPtr>
+Utility::createTagExtractors(const envoy::api::v2::Bootstrap& bootstrap) {
+
+  std::vector<Stats::TagExtractorPtr> tag_extractors;
+
+  // Ensure no tag names are repeated.
+  std::unordered_set<std::string> names;
+  auto add_tag = [&names, &tag_extractors](const std::string& name, const std::string& regex) {
+    if (!names.emplace(name).second) {
+      throw EnvoyException(fmt::format("Tag name '{}' specified twice.", name));
+    }
+
+    tag_extractors.emplace_back(Stats::TagExtractorImpl::createTagExtractor(name, regex));
+  };
+
+  // Add defaults.
+  if (!bootstrap.stats_config().has_use_all_default_tags() ||
```

Comment:
@htuch random unrelated thought: With the proto validator code, could we potentially auto add defaults for things like this if not specified? Might make things easier to read in general.